CMakeLists.txt

```diff
@@ -1,16 +1,38 @@
-cmake_minimum_required(VERSION 2.8.3)
+cmake_minimum_required(VERSION 3.5)
 project(rtcm_msgs)
 
-find_package(catkin REQUIRED COMPONENTS std_msgs message_generation)
+find_package(ros_environment REQUIRED)
 
-add_message_files(
-  FILES
-  Message.msg
-)
+set(ROS_VERSION $ENV{ROS_VERSION})
 
-generate_messages(
-  DEPENDENCIES
-  std_msgs
-)
+find_package(std_msgs REQUIRED)
 
-catkin_package(CATKIN_DEPENDS std_msgs message_runtime)
+if(${ROS_VERSION} EQUAL 1)
+
+  find_package(catkin REQUIRED
+    COMPONENTS
+      message_generation
+  )
+
+  add_message_files(DIRECTORY msg
+    FILES
+      Message.msg
+  )
+
+  generate_messages(DEPENDENCIES std_msgs)
+
+  catkin_package(CATKIN_DEPENDS message_runtime std_msgs)
+
+elseif(${ROS_VERSION} EQUAL 2)
+  find_package(ament_cmake REQUIRED)
+  find_package(builtin_interfaces REQUIRED)
+  find_package(rosidl_default_generators REQUIRED)
+
+  rosidl_generate_interfaces(${PROJECT_NAME} ADD_LINTER_TESTS
+    msg/Message.msg
+  )
```

Comment:
Looks like you missed adding the dependency for std_msgs here. It won't build for me on ROS2 (humble) without it.
I think it should be:
```cmake
rosidl_generate_interfaces(${PROJECT_NAME} ADD_LINTER_TESTS
    msg/Message.msg
    DEPENDENCIES std_msgs
  )

```

Reply:
Looks like you're right. Although it compiled just fine on my PC...
I'll add anyway to be specific.

Reply:
Yes, odd that it worked for you and not me... I tried building several times etc, but could not get it to work without explicit dependencies. 

Tested it on humble in a docker container with Ubuntu 22.04 LTS. Not sure if that matters.